From 0c357bfe708c348d2e103e1d8c065f6f690c8e5d Mon Sep 17 00:00:00 2001
From: Lorenzo Pirritano <lpirritano@anaconda.com>
Date: Mon, 9 Sep 2024 17:22:38 +0200
Subject: [PATCH 2/2] get_xticks returns list

---
 ds_utils/preprocess.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ds_utils/preprocess.py b/ds_utils/preprocess.py
index 97bdeff..d5c22e8 100644
--- a/ds_utils/preprocess.py
+++ b/ds_utils/preprocess.py
@@ -49,7 +49,7 @@ def visualize_feature(series: pd.Series, remove_na: bool = False, *, ax: Optiona
         ax.set_title(f"{feature_series.name} ({feature_series.dtype})")
         ax.set_xlabel("")
 
-    ticks_loc = ax.get_xticks().tolist()
+    ticks_loc = ax.get_xticks()
     ax.xaxis.set_major_locator(ticker.FixedLocator(ticks_loc))
     ax.set_xticklabels(labels, rotation=45, horizontalalignment='right')
 
@@ -238,7 +238,7 @@ def plot_features_interaction(feature_1: str, feature_2: str, data: pd.DataFrame
             # first feature is categorical and the second is datetime
             dup_df[feature_2] = data[feature_2].apply(dates.date2num)
             chart = sns.violinplot(x=feature_2, y=feature_1, data=dup_df, ax=ax)
-            ticks_loc = chart.get_xticks().tolist()
+            ticks_loc = chart.get_xticks()
             chart.xaxis.set_major_locator(ticker.FixedLocator(ticks_loc))
             chart.set_xticklabels(chart.get_xticklabels(), rotation=45, horizontalalignment='right')
             ax.xaxis.set_major_formatter(_convert_numbers_to_dates)
@@ -253,7 +253,7 @@ def plot_features_interaction(feature_1: str, feature_2: str, data: pd.DataFrame
             dup_df[feature_1] = data[feature_1].apply(dates.date2num)
             dup_df[feature_2] = _copy_series_or_keep_top_10(data[feature_2])
             chart = sns.violinplot(x=feature_1, y=feature_2, data=dup_df, ax=ax)
-            ticks_loc = chart.get_xticks().tolist()
+            ticks_loc = chart.get_xticks()
             chart.xaxis.set_major_locator(ticker.FixedLocator(ticks_loc))
             chart.set_xticklabels(chart.get_xticklabels(), rotation=45, horizontalalignment='right')
             ax.xaxis.set_major_formatter(_convert_numbers_to_dates)
-- 
2.39.1

